index.js

```diff
@@ -11,7 +75,9 @@ module.exports = function(req) {
 		return req;
 	}
 
-	req.set('X-Csrf-Token', localStorage['XSRF.Token']);
-	req.set('X-D2L-App-Id', 'deprecated');
+	addHeaders(req);
+
+	req.end = preflight(req);
```

Comment:
I wonder if you could avoid dealing with the `getPrototypeOf` business in `preflight` if you grabbed the existing `end` before overwriting it and passed it into `preflight`. Something like:

    var oldEnd = req.end;
    req.end = preflight( req, oldEnd );